### PR TITLE
Add overlaps method to Rect

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

**Comment by goderbauer** [comment]
googlebot: try again!

File: lib/ui/geometry.dart [hunk 1]
```diff
@@ -653,6 +653,15 @@ class Rect {
     );
   }
 
+  /// Whether `other` has any overlap with this rectangle.
```
Comment (jacob314) [inline lib/ui/geometry.dart]: Edge case nit: wouldn't it be better to ensure that two rectangles have > epsilon overlap? For example, with this code a rect with left 0 and right 50 will overlap a rectangle with left 50 and right 100? even though they there is no real overlap.

Comment (goderbauer) [reply lib/ui/geometry.dart]: It probably makes sense to define overlap as having a nonzero area of overlap. That's also how Skia seems to define overlap: https://github.com/google/skia/blob/1ffda0484d87cd44afcd7387618f4a369ba3cd23/src/gpu/GrRect.h#L59

I updated the code. PTAL.

**Comment by goderbauer** [comment]
The googlebot is having issues... After carefully double-checking that everything is alright I am submitting despite it being (wrongfully) unhappy.

**Pull request merged**